src/Orleans/Core/Grain.cs

```diff
@@ -332,5 +324,51 @@ protected virtual Task ReadStateAsync()
         {
             return storage.ReadStateAsync();
         }
+
+        void ILifecycleParticipant<GrainLifecycleStage>.Participate(ILifecycleObservable<GrainLifecycleStage> lifecycle)
```

Comment:
Implementing this explicitly would mean that app devs (or infrastructure extensions) will not be able to create a base class that is a participant and hook it up as expected, correct? Maybe it should just be protected virtual instead?